### ISSUE TITLE
[stable/rabbitmq-ha] Duplicate metrics in Prometheus

### ISSUE THREAD

**Issue by gdmello** [helm/charts#8236]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT
**Version of Helm and Kubernetes**:
```$ helm version
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
$ kc version
Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.4", GitCommit:"9befc2b8928a9426501d3bf62f72849d5cbcd5a3", GitTreeState:"clean", BuildDate:"2017-11-20T05:28:34Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"8+", GitVersion:"v1.8.4+coreos.0", GitCommit:"4292f9682595afddbb4f8b1483673449c74f9619", GitTreeState:"clean", BuildDate:"2017-11-21T17:22:25Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**:
`[stable/rabbitmq-ha]`

**What happened**:
Installing this chart with the Prometheus exporter enabled duplicates the metrics `n*n`. So, for e.g- My cluster has 5 nodes, and so have 5 exporters as well. Here's the output I see for a simple query `rabbitmq_running{namespace="prd-app", endpoint="exporter"}`-
```
rabbitmq_running{endpoint="exporter",instance="10.2.0.144:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-0.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-4",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.0.144:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-1.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-4",service="prd-app-rabbitmq-ha"} 1
rabbitmq_running{endpoint="exporter",instance="10.2.0.144:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-2.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-4",service="prd-app-rabbitmq-ha"} 1
rabbitmq_running{endpoint="exporter",instance="10.2.0.144:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-3.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-4",service="prd-app-rabbitmq-ha"} 1
rabbitmq_running{endpoint="exporter",instance="10.2.0.144:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-4.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-4",service="prd-app-rabbitmq-ha"} 1
rabbitmq_running{endpoint="exporter",instance="10.2.4.193:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-0.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-3",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.4.193:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-1.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-3",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.4.193:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-2.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-3",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.4.193:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-3.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-3",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.4.193:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-4.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-3",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.5.12:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-0.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-0",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.5.12:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-1.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-0",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.5.12:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-2.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-0",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.5.12:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-3.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-0",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.5.12:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-4.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-0",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.6.194:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-0.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-1",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.6.194:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-1.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-1",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.6.194:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-2.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-1",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.6.194:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-3.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-1",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.6.194:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-4.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-1",service="prd-app-rabbitmq-ha"}  1
rabbitmq_running{endpoint="exporter",instance="10.2.7.132:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-0.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-2",service="prd-app-rabbitmq-ha"} 1
rabbitmq_running{endpoint="exporter",instance="10.2.7.132:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-1.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-2",service="prd-app-rabbitmq-ha"} 1
rabbitmq_running{endpoint="exporter",instance="10.2.7.132:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-2.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-2",service="prd-app-rabbitmq-ha"} 1
rabbitmq_running{endpoint="exporter",instance="10.2.7.132:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-3.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-2",service="prd-app-rabbitmq-ha"} 1
rabbitmq_running{endpoint="exporter",instance="10.2.7.132:9090",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-4.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",pod="prd-app-rabbitmq-ha-2",service="prd-app-rabbitmq-ha"} 1
```
The `instance` and `pod` labels force Prometheus to treat each metric values as a distinct timeseries data point. These labels are injected by Prometheus itself to be able to distinguish the scrape source of the metric data.

**What you expected to happen**:
Running the above query, I should see only 5 results, one for each of the 5 nodes.
```
rabbitmq_running{endpoint="exporter",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-0.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",service="prd-app-rabbitmq-ha"}	1
rabbitmq_running{endpoint="exporter",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-1.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",service="prd-app-rabbitmq-ha"}	1
rabbitmq_running{endpoint="exporter",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-2.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",service="prd-app-rabbitmq-ha"}	1
rabbitmq_running{endpoint="exporter",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-3.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",service="prd-app-rabbitmq-ha"}	1
rabbitmq_running{endpoint="exporter",job="prd-app-rabbitmq-ha",namespace="prd-app",node="rabbit@prd-app-rabbitmq-ha-4.prd-app-rabbitmq-ha-discovery.prd-app.svc.cluster.local",service="prd-app-rabbitmq-ha"}	1
```

**How to reproduce it** (as minimally and precisely as possible):
Deploy with the exporter enabled.

**Anything else we need to know**:
I can work around the duplicates, via Prometheus aggregation operators, but this seems like an out-of-the-box flaw in the metrics design.

**Solution**
I've verified that using a single exporter for the entire cluster, which hooks into the Rabbitmq management port exposed by the Kubernetes `Service` works perfectly, ie only 5 rows of data are returned.

**Comment by gdmello** [comment]
@bradenwright - Since you introduced this very useful addition, did you encounter this issue?
/cc: @etiennetremel @unguiculus - your feedback is apppreciated!

**Comment by steven-sheehy** [comment]
Issue is with the [rabbitmq_exporter](https://github.com/kbudde/rabbitmq_exporter/issues/92) adding the node parameter. The exporter can be ran [standalone](https://github.com/helm/charts/tree/master/stable/prometheus-rabbitmq-exporter) instead of a sidecar like in rabbitmq-ha, hence why the need for the node label in that scenario. I've asked if they can make the node label an option.

**Comment by yoshz** [comment]
The self label is available since 1.0.0-RC6.1. 
To use this version with this helm chart use the following values:
```yaml
prometheus:
  exporter:
    enabled: true
    image:
      tag: v1.0.0-RC6.1
    port: 9419
```
